source/connection.c

```diff
@@ -672,6 +701,12 @@ struct aws_http_server *aws_http_server_new(const struct aws_http_server_options
         server->is_using_tls = true;
     }
 
+    struct aws_server_user_data *server_user_data =
+        aws_mem_calloc(options->allocator, 1, sizeof(struct aws_server_user_data));
+    server_user_data->server = server;
+    server_user_data->alloc = options->allocator;
+    server_user_data->setup_future = aws_future_void_new(options->allocator);
```

Comment:
wait a minute ... what is even the point of the `struct server_user_data`? Can't we just put the `setup_future` inside the `struct server`

EZPZ